content/blogposts/2023/copilot-vs-cody.md

```diff
@@ -0,0 +1,229 @@
+
+Our query will be: **"What does this application do?"**
+
+![Explain what a codebase does](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/explain-what-codebase-does.png)
+
+Both Cody and Copilot were able to give us an answer to our query. Cody is on the left-hand side, and Copilot is on the right, and we'll use this format throughout the post. Copilot told us about the different technologies used in the repository, while Cody told us the tech stack, APIs used, as well as the major features of the application.
+
+**Cody also cites its sources**, and we can see in the above screenshot that Cody used 14 files from our codebase to get enough context to generate a detailed response. Copilot, on the other hand, does not provide visibility into how it came up with the answer.
+
+I think we can give Cody the win in our first scenario. After round one, the score stands at:
+
+Cody - 1
+
+Copilot - 0
+
+## Scenario: Setting up the development environment
+
+Now that our developer knows a little bit about what the application does, the next logical thing for them to do is to figure out how to set up the local development environment so that they can run the application and make code changes themselves. Without a code AI assistant, the developer would again likely search for a `readme.md` file or similar instructions on how to get set up. But with Cody and Copilot ready, we can ask them how to get this application running locally.
+
+The query we'll pose is: **"How do I start this app up in development mode?"**
+
+![Setup development environment](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/setup-dev-environment.png)
+
+Both Cody and Copilot were able to give us a response to this question as well. The difference between the two responses again highlights the importance of having the right context.
+
+Cody read 12 files to generate the response and walked the developer through installing the required JavaScript and PHP dependencies, setting up an `.env` file, creating a local database, running a database migration to set up the correct tables and seed data, and finally starting the server.
+
+Copilot, on the other hand, initially told us to navigate to the project directory, but since we are already in the project directory, this step is unnecessary. Copilot then told us to install the dependencies and run the project. But there were some glaring issues. Without a proper `.env` file in the project directory, the app will not start. Not only that, but the `npm start` command, while common in Node and JavaScript projects, does not exist in this codebase. To start the local server, this codebase requires running the `npm run dev` command, which Cody correctly suggested.
```

Comment:
I would mention Copilot first, since that is the worse example and IMO it flows more naturally to go from worse to better here.

The way I'd summarize the Copilot response is "It's as if Copilot only knows that we're using JavaScript and PHP in this repository and made some very rough guesses at what commands might need to be used to set up a dev environment for those languages. Meanwhile, Cody gives more of a response you'd expect to get from an experienced engineer who actually took the time to read through the setup docs in the project.